src/util/log.d

```diff
@@ -350,18 +354,71 @@ class FileLogger(alias Layout) : Logger
     }
 }
 
-class RollingFileLogger(alias Layout) : FileLogger!Layout
+mixin template CachedAppender()
+{
+    import std.process : thisThreadID;
+
+    private static struct ThreadState
+    {
+        Appender!(char[]) appender;
+
+        int depth;
+    }
+
+    private ThreadState[typeof(thisThreadID)] threadStates;
+
+    private ref ThreadState threadState() return
+    {
+        return this.threadStates.require(thisThreadID, ThreadState(.appender!(char[]), 0));
```

Comment:
Why not use thread-local storage for the appender per thread?

Reply:
did that (I didn't think it through completely)